src/clients/Device.js

```diff
@@ -327,14 +318,18 @@ class Device extends EventEmitter {
       });*/
 
       this._sendHello();
-
       this._connectionStartTime = new Date();
 
-      // We don't need to await this as other functions will await the response
-      this._getDescription();
+      const { functionState, systemInformation } = await this._getDescription();
```

Comment:
I return functionState and systemInformation from the _getDescription promise. I strongly believe that the more void functions we have, that actually should return something(but instead they assign something to private pros inside) the less readable and unpredictable our code is.